src/Template/Bake/config/diff.ctp

```diff
@@ -0,0 +1,492 @@
+                    $columnsReference = '[' . $this->Migration->stringifyList($constraint['references'][1], ['indent' => 5]) . ']';
+                else:
+                    $columnsReference = '\'' . $constraint['references'][1] . '\'';
+                endif;
+                $statement = $this->Migration->tableStatement($tableName, true);
+                if (!empty($statement)): %>
+
+        <%= $statement %>
+                    <%- endif; %>
+            ->addForeignKey(
+                <%= $columnsList %>,
+                '<%= $constraint['references'][0] %>',
+                <%= $columnsReference %>,
+                [
+                    'update' => '<%= $constraint['update'] %>',
+                    'delete' => '<%= $constraint['delete'] %>'
+                ]
+            )
+                <%- endif; %>
+            <%- endforeach; %>
+                <%- if (isset($this->Migration->tableStatements[$table])): %>
+            ->update();
+            <%- endif; %>
+        <%- endif; %>
+        <%- foreach ($data as $tableName => $tableDiff): %>
+            <%- if (!empty($tableDiff['indexes']['add'])): %>
+
+        $this->table('<%= $tableName %>')
+                <%- foreach ($tableDiff['indexes']['add'] as $indexName => $index): %>
+            ->removeIndexByName('<%= $indexName %>')
+                <%- endforeach %>
+            ->update();
+            <%- endif; %>
+            <%- if (!empty($tableDiff['indexes']['remove'])): %>
+
+        $this->table('<%= $tableName %>')
+            <%- foreach ($tableDiff['indexes']['remove'] as $indexName => $indexDefinition): %>
+            ->addIndex(
+                [<% echo $this->Migration->stringifyList($indexDefinition['columns'], ['indent' => 5]); %>],
+                [<%
+                $params = ['name' => $indexName];
+                if ($indexDefinition['type'] === 'unique'):
+                    $params['unique'] = true;
+                endif;
+                echo $this->Migration->stringifyList($params, ['indent' => 5]);
+                %>]
+            )
+            <%- endforeach; %>
+            ->update();
+        <%- endif; %>
+        <%- if (!empty($tableDiff['columns']['remove'])): %>
+
+        $this->table('<%= $tableName %>')
+        <%- foreach ($tableDiff['columns']['remove'] as $columnName => $columnDefinition):
+                $type = $columnDefinition['type'];
+                unset($columnDefinition['type']);
+                $columnOptions = $this->Migration->getColumnOption($columnDefinition);
+                $columnOptions = $this->Migration->stringifyList($columnOptions, ['indent' => 4]);
+            %>
+            ->addColumn('<%= $columnName %>', '<%= $type %>', [<%= $columnOptions %>])
+        <%- endforeach; %>
+            ->update();
+        <%- endif; %>
+        <%- if (!empty($tableDiff['columns']['changed'])):
+            $oldTableDef = $dumpSchema[$tableName];
+            %>
```

Comment:
Agreed, that is part of the plan.
On top of that, the diff and snapshot bake command now share some similarities.